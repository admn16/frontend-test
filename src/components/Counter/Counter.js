import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledCounter = styled.div`
  align-items: center;
  display: flex;
  justify-content: space-between;
  margin: 5px 0;

  &:hover {
    background-color: #f7fcfd;
  }
`;

const Title = styled.span`
  color: #312a2a;
  flex-grow: 10;
  font-size: 15px;
  max-width: 250px;
  overflow: hidden;
  text-overflow: ellipsis;
`;

const Count = styled.span`
  font-size: 14px;
  text-align: center;
  width: 40px;
`;

const Delete = styled.button`
  background: none;
  border: none;
  color: #355869;
  cursor: pointer;
  font-size: 10px;
  margin: 0 10px 0 5px;
  outline: none;
  
  &:hover {
    color: #d01204;
  }
`;

const Increment = styled(Delete)`
  color: #d01204;
  font-size: 12px;
  margin: 0;
  
  &:hover {
    color: #ff1100;
  }
`;

const Decrement = styled(Increment)`
  color: #4791a0;

  &:hover {
    color: #4dcbe4;
  }
`;

const Counter = ({
  count,
  title,
  onDelete,
  onIncrement,
  onDecrement,
}) => (
  <StyledCounter>
    <Delete type="button" onClick={onDelete}>
      <i className="fas fa-trash-alt" />
    </Delete>

    <Title>{ title }</Title>

    <Increment type="button" onClick={onIncrement}>
      <i className="fas fa-plus" />
    </Increment>

    <Count>{ count }</Count>

    <Decrement type="button" onClick={onDecrement}>
      <i className="fas fa-minus" />
    </Decrement>
  </StyledCounter>
);

Counter.propTypes = {
  count: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  onDelete: PropTypes.func.isRequired,
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
};

export default Counter;
