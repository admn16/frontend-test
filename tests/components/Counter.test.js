import React from 'react';
import renderer from 'react-test-renderer';
import Counter from 'components/Counter/Counter';

describe('<Counter/>', () => {
  const mockFn = jest.fn();
  const props = {
    count: 0,
    title: 'Title',
    onDelete: mockFn,
    onIncrement: mockFn,
    onDecrement: mockFn,
  };

  test('renders correctly', () => {
    const tree = renderer
      .create(<Counter {...props} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
