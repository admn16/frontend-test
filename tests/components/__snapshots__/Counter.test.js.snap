// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter/> renders correctly 1`] = `
<div
  className="Counter__StyledCounter-sc-1q55mxl-0 eWqOki"
>
  <button
    className="Counter__Delete-sc-1q55mxl-3 foqxoZ"
    onClick={[MockFunction]}
    type="button"
  >
    <i
      className="fas fa-trash-alt"
    />
  </button>
  <span
    className="Counter__Title-sc-1q55mxl-1 gzIbEF"
  >
    Title
  </span>
  <button
    className="Counter__Increment-sc-1q55mxl-4 gSvDCI Counter__Delete-sc-1q55mxl-3 foqxoZ"
    onClick={[MockFunction]}
    type="button"
  >
    <i
      className="fas fa-plus"
    />
  </button>
  <span
    className="Counter__Count-sc-1q55mxl-2 egAVVf"
  >
    0
  </span>
  <button
    className="Counter__Decrement-sc-1q55mxl-5 cJKjoF Counter__Increment-sc-1q55mxl-4 gSvDCI Counter__Delete-sc-1q55mxl-3 foqxoZ"
    onClick={[MockFunction]}
    type="button"
  >
    <i
      className="fas fa-minus"
    />
  </button>
</div>
`;
